### PR TITLE
Replace Enums with dict and conversion functions

### PR DESCRIPTION
**Pull request by guynir42**
First step: replace image type and file formats with dictionaries and converters. 
Next: to add a "bad mask" with a bit flag to all data types.

### CODE REVIEW

**Comment by guynir42** [comment]
Addresses #69

File: models/enums_and_bitflags.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+    25: 'Bad Pixel',
+    26: 'Bleed Trail',
+}
+cutouts_badness_inverse = {v.lower().replace(' ', ''): k for k, v in cutouts_badness_dict.items()}
+
+source_list_badness_dict = {
+    41: 'X-Match Failed',
+    42: 'Big Residuals',
+    43: 'Few Stars',
+    44: 'Many Stars',
```
Comment (rknop) [inline models/enums_and_bitflags.py:227]: Too Many Stars?

Many stars doesn't sound like a bad thing to me.

File: models/enums_and_bitflags.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+    11: 'avro',
+    12: 'netcdf',
+    13: 'jpg',
+    14: 'png',
+    15: 'pdf',
+}
+
+allowed_image_formats = ['fits', 'hdf5']
+image_format_dict = {k: v for k, v in file_format_dict.items() if v in allowed_image_formats}
+image_format_inverse = {v: k for k, v in image_format_dict.items()}
```
Comment (rknop) [inline models/enums_and_bitflags.py:27]: Random musing : it's too bad python doesn't have a symmetric 1:1 mapping class.  As far as  I know, having an inverse dictionary is the only reasonable way to do it.

File: models/enums_and_bitflags.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    44: 'Many Stars',
+}
+source_list_badness_inverse = {v.lower().replace(' ', ''): k for k, v in source_list_badness_dict.items()}
+
+# join the badness:
+data_badness_dict = {0: 'Good'}
+data_badness_dict.update(image_badness_dict)
+data_badness_dict.update(cutouts_badness_dict)
+data_badness_dict.update(source_list_badness_dict)
+data_badness_inverse = {v.lower().replace(' ', ''): k for k, v in data_badness_dict.items()}
```
Comment (rknop) [inline models/enums_and_bitflags.py:236]: To catch future programming errors, it might be worth adding a test to make sure that none of (image_badness_dict, cutouts_badness_dict, source_list_badness_dict) (and anything we add later) have 0 as a key, and that none of them have overlapping keys with each other.

File: models/enums_and_bitflags.py (hunk 1 above)
Comment (guynir42) [reply models/enums_and_bitflags.py:227]: I agree to many sources is better. I think this sort of thing could happen when SExtractor fails and thinks every pixel is a source.

File: models/enums_and_bitflags.py (hunk 2 above)
Comment (guynir42) [reply models/enums_and_bitflags.py:27]: We only need to calculate it once here, so it is not so bad.

File: models/enums_and_bitflags.py (hunk 3 above)
Comment (guynir42) [reply models/enums_and_bitflags.py:236]: I agree about testing against zeros. 
Overlaps in principle are not illegal. I can imagine places where different models could have overlapping keywords.

Comment (rknop) [reply models/enums_and_bitflags.py:236]: But the dict that is used for lookup will only have the latest one that got updated

Comment (guynir42) [reply models/enums_and_bitflags.py:236]: So you're worried that someone might alter the dict during runtime? We can probably make up a bidirectional dictionary for fun.

Comment (guynir42) [reply models/enums_and_bitflags.py:236]: and of course this has already been implemented: https://pypi.org/project/bidict/

**Comment by guynir42** [comment]
@rknop this one is really hacky but in the end it works. I am not sure how all of these crazy outer joins will scale once we have lots of images, but I hope it doesn't end up growing the number of rows a lot more than the total number of images. 

Also, I'm leaving the implementation for Cutouts for a different PR, when I start looking at that model more closely.

**Comment by guynir42** [comment]
Addresses #37 #40 #73

File: models/enums_and_bitflags.py (hunk 1 above)
Comment (rknop) [reply models/enums_and_bitflags.py:273]: I'd still recommend changing this to "too many sources" and "too few sources"

Comment (guynir42) [reply models/enums_and_bitflags.py:273]: Yes, I will do that. The beauty of this new setup is that we can make all the changes we want down the line. There's no need to commit to these enums at all. In fact we should probably discuss which values we want and how to name them.

**Comment by guynir42** [comment]
@rknop I fixed the two comments on the Enums. It is worth while that you also look at the bitflag mechanics. That's the really complicated one and I'm sure you'll have a lot to say about it...

**Review by rknop — APPROVED** [review]
This looks good.  Go ahead and merge it once the alembic two-heads issue is resolved.
